### PR TITLE
Fixed #1390: Restructure theme to match MUI theming

### LINKED ISSUE #1390: restructure current theme.js to follow MUI theme structure.
**What would you like to be added**:
A matching theming attributes to MUI default theme structure
Change our current theming from ...
```
const theme = createMuiTheme({
  palette: {
    primary: {
      main: '#333E64',
    },
    secondary: {
      main: '#0589D6',
    },
    error: {
      main: red.A400,
    },
    background: {
      default: '#E5E5E5',
    },
    text: {
      primary: '#E5E5E5',
      secondary: '#8BC2EB',
      default: '#181818',
    },
  },
});
```
to Something similar to the following...

image

Luckily it seems that we only have a conflict in text attribute, there is no `default` key in `text` 
**Why would you like this to be added**:
This will help with Typescript when we migrate to Nextjs and allow light/dark mode switching easily

### PR DESCRIPTION
**Pull request by tonyvugithub**
## Issue This PR Addresses
Fixed #1390 

## Type of Change

- [x] **Bugfix**: Change which fixes an issue
- [ ] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description
Thanks to conversation with @agarcia-caicedo yesterday. We agree that we can change the color of `default` in `text` attribute to `primary` and make the color of `primary` to `contrastText` under `primary` property instead since it is being used for contrast text right now. With this change, when we switch to theming in TS, TS won't complain as `contrastText`  is one options in `palette/primary`. I also updated all the components with the change in theming too. The look is the same with the new change. 

## Checklist

- [ ] **Quality**: This PR builds and passes our npm test and works locally
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

**Review by humphd — CHANGES_REQUESTED** [review]
This needs needs to be documented.  We have a bunch of PRs up right now that will undo what you have here, so we need a doc we can point people at that explains how to use the theme colours.  I would suggest that we rename `https://github.com/Seneca-CDOT/telescope/blob/master/docs/gui-colour-palette.md` to `theme-and-colours.md` and add detailed info about how to use the theme colours vs. just what they are.

Also, I think `text.contrastText` is redundant, and `text.contrast` is sufficient.

**Comment by tonyvugithub** [comment]
@humphd : I will update the docs shortly.

**Comment by tonyvugithub** [comment]
@humphd : Do you suggest me to wait until all the frontend PRs got merged before doing this?

**Comment by humphd** [comment]
@tonyvugithub that might never happen :)  I think we better get your docs in, land this change, and then be careful with reviews.

File: docs/theme-and-colours.md
Comment (humphd) [inline docs/theme-and-colours.md:77]: `contrast` vs. `contrastText` here

File: docs/theme-and-colours.md
Comment (humphd) [inline docs/theme-and-colours.md:76]: If this is a background colour, would `primary.background` be more intuitive?  Or is `main` a thing with MUI?

File: docs/theme-and-colours.md
Comment (humphd) [inline docs/theme-and-colours.md:78]: Same question here, `background` vs. `main`?

File: docs/theme-and-colours.md
Comment (humphd) [inline docs/theme-and-colours.md:81]: So now we have `primary.text` and `text.primary`?  This is going to confuse people, including me.

**Review by humphd — CHANGES_REQUESTED** [review]

File: docs/theme-and-colours.md
Comment (tonyvugithub) [reply docs/theme-and-colours.md:76]: @humphd : main is a thing with MUI. I pushed another commit.
MUI theme looks like this 
image

File: docs/theme-and-colours.md
Comment (tonyvugithub) [reply docs/theme-and-colours.md:78]: @humphd : everything I have here is following MUI theming

File: docs/theme-and-colours.md
Comment (tonyvugithub) [reply docs/theme-and-colours.md:81]: @humphd : I can twist MUI theme by overwriting their default theme using our own definitions and avoid TS complaint

File: docs/theme-and-colours.md
Comment (tonyvugithub) [reply docs/theme-and-colours.md:76]: `contrastText` was used due to following what in MUI theming

File: docs/theme-and-colours.md
Comment (humphd) [reply docs/theme-and-colours.md:77]: Interesting.  In that case, we should leave it.  What bizarre naming choices!

**Comment by tonyvugithub** [comment]
@humphd : Do I have your approval on this?

**Comment by humphd** [comment]
Needs a rebase

**Review by humphd — APPROVED** [review]

**Review by agarcia-caicedo — APPROVED** [review]

**Pull request merged**
